crates/subspace-farmer/src/rpc.rs

```diff
@@ -21,9 +22,9 @@ pub trait RpcClient {
         block_number: u32,
     ) -> Result<Option<EncodedBlockWithObjectMapping>, Error>;
 
-    async fn subscribe_new_head(&self) -> Result<Subscription<NewHead>, Error>;
+    async fn subscribe_new_head(&self) -> Result<Receiver<NewHead>, Error>;
```

Comment:
You still have `jsonrpsee`-specific error type here though...

Reply:
Yes, I was not sure about abstracting it. Will do